src/Apps/ArtAdvisor/04-Bio-Follows-Alerts/lib/functions.ts

```diff
@@ -0,0 +1,202 @@
+import chalk from "chalk"
+import { metaphysics } from "Apps/ArtAdvisor/04-Bio-Follows-Alerts/lib/metaphysics"
+
+/*
+ * Get the user's artsy profile as returned from the "me" query.
+ */
+
+export async function getUserProfile(args: { size: number; token: string }) {
+  const query = `query getUserProfile {
+    me {
+      internalID
+      name
+      email
+      followsAndSaves {
+        artistsConnection(first: 10) {
+          edges {
+            node {
+              artist {
+                name
+              }
+            }
+          }
+        }
+        genesConnection(first: 10) {
+          edges {
+            node {
+              gene {
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  }`
+
+  const variables = {
+    size: args.size,
+  }
+
+  const headers = {
+    "X-ACCESS-TOKEN": args.token,
+    "Content-Type": "application/json",
+  }
+
+  const response = await metaphysics({ query, variables, headers })
+
+  console.log(
+    chalk.yellow("createAlert response: "),
+    JSON.stringify(response, null, 2)
+  )
+
+  const profileOrError = response.data.me || response.errors
```

Comment:
Attempts to replicate the responseOrError pattern that _most_ our mutations respond with. Maybe a bit awkward, but kind of a work around to not having better validation or error handling at this stage. There _should not_ be anything coming back from a response that would problematic to feed back to the model.

Reply:
I don't know how applicable it is here, but note that by using `fetchQuery` via relay, we get store caching around query/var reuse. This should speed things up where there are repeat hits to the same combination of factors.